chia/_tests/core/data_layer/test_data_store.py

```diff
@@ -1854,30 +1553,34 @@ async def test_insert_from_delta_file(
             store_id=store_id,
             status=Status.COMMITTED,
         )
+        await data_store.add_node_hashes(store_id)
 
     root = await data_store.get_tree_root(store_id=store_id)
-    assert root.generation == num_files + 1
+    assert root.generation == num_files
```

Comment:
It seems like changes to this ought to be separate from changing out the data store backend?

Reply:
I think it was off by one before (we did an unnecessary `create_tree`, introducing another empty root at generation=1)